controller/src/controller/models/domain/rsptag.py

```diff
@@ -402,10 +511,13 @@ def _compare(self, other: object) -> int:
             0 if equal, -1 if self is less than other, 1 if self is greater
             than other, `NotImplemented` if they're not comparable.
         """
-        if not isinstance(other, RSPImageTag):
-            return NotImplemented
-        if self.image_type != other.image_type:
+        try:
+            self._check_comparable_types(other)
+        except NotImplementedError:
             return NotImplemented
+        # If we got this far, we know they are both RSPImageTags even if
+        # mypy doesn't.
+        other = cast("RSPImageTag", other)
```

Comment:
If you're going to move the check to a method, you need to use a type guard so that the type inference works. That will avoid the cast and ensure the typing still works properly. See [the mypy documentation](https://mypy.readthedocs.io/en/stable/type_narrowing.html#user-defined-type-guards), the short version of which is in the suggestion below on the helper function.